plugins/pixart-rf/fu-pxi-firmware.c

```diff
@@ -130,10 +132,24 @@ fu_pxi_firmware_parse(FuFirmware *firmware,
 	fu_dump_raw(G_LOG_DOMAIN, "fw header", fw_header, sizeof(fw_header));
 
 	/* set fw version */
-	version_raw = (((guint32)(fw_header[0] - '0')) << 16) +
-		      (((guint32)(fw_header[2] - '0')) << 8) + (guint32)(fw_header[4] - '0');
+	if (fu_pxi_firmware_is_hpac(self)) {
+		if (!fu_memread_uint16_safe(buf,
+					    bufsz,
+					    bufsz - PIXART_RF_FW_HEADER_HPAC_VERSION_POS_FROM_END,
+					    &hpac_ver,
+					    G_BIG_ENDIAN,
+					    error))
+			return FALSE;
+
+		version_raw = hpac_ver;
+		version = fu_pxi_hpac_version_info_parse(hpac_ver);
```

Comment:
I think you need an #include `#include "fu-pxi-common.h"` at the top of the file for this.